app/code/Magento/Inventory/Model/SourceCarrierLink.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Inventory\Model;
+
+use Magento\Framework\Model\AbstractModel;
+use \Magento\InventoryApi\Api\Data\SourceCarrierLinkInterface;
+
+class SourceCarrierLink extends AbstractModel implements SourceCarrierLinkInterface
+{
+
+    /**
+     * Name of the resource collection model
+     *
+     * @codingStandardsIgnore
+     * @var string
+     */
+    protected $_collectionName = 'Magento\Inventory\Model\Resource\SourceCarrierLink\Collection';
```

Comment:
it's better to use _init method in pseudo constructor instead of initializing this variable 

```php
        $this->_resourceName = $resourceName;
        if ($collectionName === null) {
            $collectionName = $resourceName . '\\' . 'Collection';
        }
        $this->_collectionName = $collectionName;
```
thus we have a convention to name a Collection the same way as Resource model with adding suffix Collection